conformance/grpc-echo/grpcechoserver.go

```diff
@@ -0,0 +1,253 @@
+/*
+Copyright 2024 The Kubernetes Authors.
```

Comment:
Is all of this net new or is it coming from somewhere else?

Reply:
This is adapted from https://github.com/kubernetes-sigs/ingress-controller-conformance/pull/106 . Still 100% authored by me, but technically in 2023. Since it was never checked in, I figured 2024 was more appropriate.